src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -1761,10 +1770,34 @@ private void VisitQueryableMethodCall(MethodCallExpression expression)
 
                     VisitExpression(expression.Arguments[0]);
                     break;
+                case "Filter":
+                {
+                    _insideFilter++;
+                    VisitExpression(expression.Arguments[0]);
+                    FilterMode(@on: true);
+                    if (_chainedWhere)
+                    {
+                        DocumentQuery.AndAlso();
+                        DocumentQuery.OpenSubclause();
+                    }
+                    if (_chainedWhere == false && _insideFilter > 1)
+                        DocumentQuery.OpenSubclause();
+                    
+                    VisitExpression(((UnaryExpression)expression.Arguments[1]).Operand);
+                    FilterMode(@on: true);
```

Comment:
Didn't we call this on line 1777 ?

Reply:
Changed into `FilterModeScope` and removed.